go/arrow/numeric/float16.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package numeric
```

Comment:
I am not too fond of this package name.
I don't have many alternative naming that I really like either, though.

```go
package float16
package flt16
package f16

type Float uint16
type Float16 uint16
type Value uint16
type F16

func New(f float32) ... { ... }
```

I know it's veering (too?) closely to bikeshedding...

I'd slightly err on:
```go
package f16

type Float  uint16
type Floats []Float

func New(v Float16)      Float  { ... }
func From(vs ...float32) Floats { ... }
// or:
func From(vs []float32) Floats { ... }
```

that would look like this, from a user perspective:
```go
var v16 f16.Float
var v32 = f16.New(42).Float()
var v32s = f16.From(1, 2, 3).Floats()
var v32s = f16.From(slice...).Floats()
// or:
var v32s = f15.From([]float32{1, 2, 3}).Floats()
```

WDYT? @alexandreyc @stuartcarnie ?

Reply:
My preference would be to name the package `float16` and the type `Float16` to make things very explicit though repetitive.